contextlib.contextmanager does not permit type variance in subclasses

**Bug Report**

When using `contextlib.contextmanager`, type variance on the generator is not permitted.

**To Reproduce**

Example code:

```py3
from contextlib import contextmanager
from typing import Generator, Iterator


class ObjBase:
    pass


class ObjSub(ObjBase):
    pass


class Base:
    @contextmanager
    def cm(self) -> Generator[ObjBase, None, None]:
        raise NotImplementedError

    @contextmanager
    def cmi(self) -> Iterator[ObjBase]:
        raise NotImplementedError

    def gen(self) -> Generator[ObjBase, None, None]:
        raise NotImplementedError

    def it(self) -> Iterator[ObjBase]:
        raise NotImplementedError


class Sub(Base):
    @contextmanager
    def cm(self) -> Generator[ObjSub, None, None]:  # Error.
        yield ObjSub()

    @contextmanager
    def cmi(self) -> Iterator[ObjSub]:  # Error.
        yield ObjSub()

    def gen(self) -> Generator[ObjSub, None, None]:
        yield ObjSub()

    def it(self) -> Iterator[ObjSub]:
        yield ObjSub()
```


**Expected Behavior**

No errors from mypy.

**Actual Behavior**

```
tmp.py:31: error: Return type "_GeneratorContextManager[ObjSub]" of "cm" incompatible with return type "_GeneratorContextManager[ObjBase]" in supertype "Base"
tmp.py:35: error: Return type "_GeneratorContextManager[ObjSub]" of "cmi" incompatible with return type "_GeneratorContextManager[ObjBase]" in supertype "Base"
```

**Your Environment**

- Mypy version used: `mypy 0.790`
- Mypy command-line flags:  `$ mypy --config-file=mypy.ini ./**/*.py`
- Mypy configuration options from `mypy.ini` (and other config files):

    ```
    [mypy]
    python_version = 3.8
    ; warn_return_any = True
    warn_unused_configs = True
    ignore_missing_imports = True
    cache_dir = /dev/null
    ```

- Python version used: `Python 3.8.5`
- Operating system and version: (`$ uname -vrm`) `5.4.0-52-generic python/mypy#57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64`

--- a/stdlib/2and3/contextlib.pyi
+++ b/stdlib/2and3/contextlib.pyi
@@ -11,6 +11,7 @@ if sys.version_info >= (3, 7):
     AbstractAsyncContextManager = AsyncContextManager
 
 _T = TypeVar("_T")
+_T_co = TypeVar("_T_co", covariant=True)
 _T_io = TypeVar("_T_io", bound=Optional[IO[str]])
 _F = TypeVar("_F", bound=Callable[..., Any])
 
@@ -18,12 +19,12 @@ _ExitFunc = Callable[[Optional[Type[BaseException]], Optional[BaseException], Op
 _CM_EF = TypeVar("_CM_EF", ContextManager[Any], _ExitFunc)
 
 if sys.version_info >= (3, 2):
-    class _GeneratorContextManager(ContextManager[_T], Generic[_T]):
+    class _GeneratorContextManager(ContextManager[_T_co]):
         def __call__(self, func: _F) -> _F: ...
     def contextmanager(func: Callable[..., Iterator[_T]]) -> Callable[..., _GeneratorContextManager[_T]]: ...
 
 else:
-    class GeneratorContextManager(ContextManager[_T], Generic[_T]):
+    class GeneratorContextManager(ContextManager[_T_co]):
         def __call__(self, func: _F) -> _F: ...
     def contextmanager(func: Callable[..., Iterator[_T]]) -> Callable[..., ContextManager[_T]]: ...
 
@@ -33,7 +34,7 @@ if sys.version_info >= (3, 7):
 if sys.version_info < (3,):
     def nested(*mgr: ContextManager[Any]) -> ContextManager[Iterable[Any]]: ...
 
-class closing(ContextManager[_T], Generic[_T]):
+class closing(ContextManager[_T]):
     def __init__(self, thing: _T) -> None: ...
 
 if sys.version_info >= (3, 4):

